crates/configuration/src/version5/introspection.sql

```diff
@@ -1495,7 +1495,7 @@ WITH
     WITH
       comparison_infix_operators_mapped AS
       (
-        SELECT
+        SELECT DISTINCT ON (map.exposed_name, op.*)
```

Comment:
You'd need to update the introspection query of v3 and v4 too if you make changes to them.

IIRC The code has to solve a similar disambiguation problem in the section that deals with implicit cast extension. There it uses LIMIT + ROW_NUMBER to emulate DISTINCT ON + ORDER BY, because cockroachdb just throws ordering of nested queries out the window (see the commentary there)

Reply:
Actually I think we should leave v3 and v4 completely untouched. We shouldn't service them at all unless for a security issue.

Reply:
Removed the distinct on as we agreed that it's not necessary because each database only defines one variant of the neq operator.